test/fixtures/async.js

```diff
@@ -0,0 +1,6 @@
+module.exports = function(app, cb) {
+  // ensure boot script executes asynchronously
+  app.asyncBoot = typeof cb === 'function';
```

Comment:
This does not really verify that the script are executed asynchronously. You should check that the boot process waits until the async scripts finished.

I think you should rework the two new tests and merge them into one.

```js
// async.js
module.exports = function(app, cb) {
  app.asyncBoot = typeof cb === 'function';
  process.nextTick(cb);
}

// in the test
var app = require(...);
expect(app.booting, 'app.booting').to.be.true();
app.on('booted', function() {
  expect(app.asyncBoot, 'app.asyncBoot').to.be.true();
  done(); 
});
// the test will time out if `booted` is not emitted
```

Thoughts?